0013-improve-package-publish-ux/README.md

```diff
@@ -0,0 +1,290 @@
+
+Each ZEP covers one "feature" or "enhancement" throughout its lifecycle. You don’t
+need a new ZEP for moving from beta to GA. If new details emerge, edit the existing
+ZEP. Once a feature is "implemented", major changes should go in new ZEPs.
+
+The latest instructions for this template can be found in [this repo](/NNNN-zep-template/README.md).
+
+**Note:** PRs to move a ZEP to `implementable`, or significant changes to an
+`implementable` ZEP, must be approved by all ZEP approvers. If an approver is no
+longer appropriate, updates to the list must be approved by the remaining approvers.
+-->
+
+# ZEP-0013: Improve Package Publish UX
+
+<!--
+Keep the title short simple and descriptive. It should clearly convey what
+the ZEP is going to cover.
+-->
+
+<!--
+A table of contents helps reviewers quickly navigate the ZEP and highlights
+any additional information provided beyond the standard ZEP template.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+      - [Unit tests](#unit-tests)
+      - [e2e tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Summary
+
+<!--
+This section is key for creating high-quality, user-focused documentation
+like release notes or a roadmap. You should gather this info before
+implementation starts to keep the focus on development, not writing. ZEP
+editors should ensure the `Summary` is clear and useful for a broad audience.
+
+A good summary should be at least a paragraph long.
+
+Follow the [documentation style guide] for this section and the rest of the ZEP.
+Keep line lengths reasonable to make it easier for reviewers to provide
+feedback and reduce unnecessary changes.
+
+[documentation style guide]: https://docs.zarf.dev/contribute/style-guide/
+-->
+
+`zarf package publish` accepts either a directory with a `zarf.yaml` file or a created Zarf package. The behavior when accepting a directory is to create and publish a [skeleton package](https://docs.zarf.dev/faq/#what-is-a-skeleton-zarf-package). This is not intuitive, the average user would most likely expect a package to be built then published. They likely wouldn't know what a skeleton package is as well. Instead we will make an explicit command `zarf package publish skeleton` for skeleton package users.
```

Comment:
If you're claiming that users don't know what a skeleton package is, why not better explain what it is and how that works, rather than introducing entirely separate command, with the skeleton name in it? IMO, the improved UX in that case is questionable. Maybe rather than calling this a skeleton package, we can just say a directory containing zarf.yaml?

Reply:
Ideally, link to https://docs.zarf.dev/faq/#what-is-a-skeleton-zarf-package

Reply:
I agree we should add this link, or at least another hook for users to better understand skeleton packages. And I think this is something we can do now whether or not this ZEP is merged. Still, I am generally not a fan of the overloaded nature this puts the publish command into. The result of the publish is entirely different when it receives a directory vs when it receives a package.

Reply:
Another reason behind my motivation here is that I don't believe skeleton packages are a particularly well thought out feature. They do solve a problem of making custom init packages easier to maintain, but besides that I'm not sure I've heard of anyone using them. I've also found bugs in skeleton packages that have been around for a while and were never reported. Will add this to the motivation section.